tests/Integration/Behaviour/Features/Context/Domain/ProductFeatureContext.php

```diff
@@ -55,6 +63,158 @@ public function setProductTaxRulesGroup(string $taxRulesGroupReference, string $
+    {
+        $data = $table->getRowsHash();
+        $defaultLangId = (int) Configuration::get('PS_LANG_DEFAULT');
+
+        try {
+            $productId = $this->getCommandBus()->handle(new AddProductCommand(
+                [$defaultLangId => $data['name']],
+                $this->getProductTypeValueByName($data['type'])
+            ));
+
+            $this->getSharedStorage()->set($productReference, $productId->getValue());
+        } catch (Exception $e) {
+            $this->lastException = $e;
+        }
+    }
+
+    /**
+     * @Then /^product "(.+)" name should be "(.+)" in default language$/
+     *
+     * @param string $productReference
+     * @param string $name
+     */
+    public function assertProductNameInDefaultLang(string $productReference, string $name)
+    {
+        $defaultLangId = (int) Configuration::get('PS_LANG_DEFAULT');
+        $product = $this->getProductByReference($productReference);
+
+        if ($product->name[$defaultLangId] === $name) {
+            return;
+        }
+
+        throw new RuntimeException(
+            sprintf(
+                'Expected product name to be "%s" in default language, but it is "%s"',
+                $name,
+                $product->name[$defaultLangId]
+            )
+        );
+    }
+
+    /**
+     * @Then product :productReference should have following values:
+     *
+     * @param string $productReference
+     * @param TableNode $table
+     */
+    public function assertProductFields(string $productReference, TableNode $table)
+    {
+        $product = $this->getProductByReference($productReference);
+        $data = $table->getRowsHash();
+
+        if (!empty($data['active'])) {
+            $status = PrimitiveUtils::castStringBooleanIntoBoolean($data['active']);
+            $statusInWords = $status ? 'enabled' : 'disabled';
+
+            if ((bool) $product->active !== $status) {
+                throw new RuntimeException(sprintf('Product expected to be %s', $statusInWords));
+            }
+        }
+
+        if (!empty($data['condition'])) {
+            if ($product->condition !== $data['condition']) {
+                throw new RuntimeException(sprintf(
+                    'Product condition expected to be "%s, but is "%s"',
+                    $data['condition'],
+                        $product->condition
+                    )
+                );
+            }
+        }
+    }
+
+    /**
+     * @Then product :productReference should be assigned to default category
+     *
+     * @param string $productReference
+     */
+    public function assertProductAssignedToDefaultCategory(string $productReference)
+    {
+        $defaultCategoryId = (int) Configuration::get('PS_HOME_CATEGORY');
+        $product = $this->getProductByReference($productReference);
+        $productCategories = $product->getCategories();
+
+        foreach ($productCategories as $categoryId) {
```

Comment:
This loop validates that the first category is the default one, but if it's the second one it will fail You can use a boolean instead that sotres if the default category was found, and throw the exception when it remains false

Reply:
lol really :sweat_smile: :+1: